lib/src/network_request.dart

```diff
@@ -154,4 +177,9 @@ class NetworkRequestBody with _$NetworkRequestBody {
 
   /// A JSON body. Passed directly to `data` of the request.
   const factory NetworkRequestBody.json(Json data) = _Json;
+
+  /// A raw body. Allows for nullable untyped data that is passed directly
+  /// to `data` of the request, useful for instances where the data type
+  /// is not known until runtime.
```

Comment:
May be worth noting how `data` will be treated? As in, do we just call `toString()` on it and send it as the body? Does it have to conform to something that can be `.toJson`ed?

Reply:
Does:

> nullable untyped data that is passed directly to `data` of the request

not help?

`Dio`s data is untyped (`Object?`) so the type here can be left as is.

Reply:
perhaps related to Will's line of questioning: does the usage of the `.json` constructor here result in setting the content-type header to application/json whereas this type will result in _not_ setting the header? the reason i ask is because technically if we're sending json we are supposed to set that header and if we're sending something else we should set the corresponding header for that data type (form-url-encoded, mutlipart, application/xml, whatever)

Reply:
We don't do any of that (mainly because `Dio` does it for you). It may have been that I originally planned to, but then learned `Dio` did it (hard to recall now). If we don't feel there's a big enough value-add from the type safety/readability offered by these (I don't imagine consumers are passing `NetworkRequestBody`s around, so type safety is maybe not a huge deal here), we could nix the type entirely and just allow `Object?` 🤷‍♂️

Reply:
nah, i like having the type safety for specifying the interface. I wouldn't won't to accept `Object?` where a more constrained type would be better

Reply:
Worth calling out that we actually remove the implied content type interceptor by default, so consumers of `sturdy` need to do this themselves. I should probably add documentation on the `NetworkRequestBody` type at the very least indicating that this doesn't dictate content type deduction.

Reply:
Follow-up PR here: https://github.com/Betterment/sturdy_http/pull/4